src/OrchardCoreContrib.ContentPreview/PagePreviewMiddleware.cs

```diff
@@ -80,7 +80,11 @@ public Task Invoke(HttpContext context)
 
             if (!path.Contains("preview") && !isPreview)
             {
-                context.Response.Redirect($"/{_shellSettings.RequestUrlPrefix}/preview{path}");
+                var url = _shellSettings.Name == "Default"
+                    ? "preview" + path
+                    : $"/{_shellSettings.RequestUrlPrefix}/preview{path}";
```

Comment:
Tenants can use custom hosts too. Why don't you create the URL with `UrlHelper`?

Reply:
I will try to test that too, thanks